src/main/java/com/sequenceiq/cloudbreak/service/stack/flow/ProvisionContext.java

```diff
@@ -88,33 +89,46 @@ public void buildStack(final CloudPlatform cloudPlatform, Long stackId, Map<Stri
                         CreateResourceRequest createResourceRequest =
                                 resourceBuilder.buildCreateRequest(pCO,
                                         Lists.newArrayList(resourceSet),
-                                        resourceBuilder.buildResources(pCO, 0, Arrays.asList(resourceSet)), 0);
+                                        resourceBuilder.buildResources(pCO, 0, Arrays.asList(resourceSet),
+                                                Lists.newArrayList(stack.getInstanceGroups()).get(0)),
+                                        0,
+                                        Lists.newArrayList(stack.getInstanceGroups()).get(0));
```

Comment:
It seems like really strange here to pass the first item of the instanceGroup array (that is converted from a Set so its items are randomly ordered I think). Use null if it's not used anywhere.